pkg/awsutils/awsutils.go

```diff
@@ -755,6 +769,11 @@ func (cache *EC2InstanceMetadataCache) AllocENI(useCustomCfg bool, sg []*string,
 	_, err = cache.ec2SVC.ModifyNetworkInterfaceAttributeWithContext(context.Background(), attributeInput)
 	awsAPILatency.WithLabelValues("ModifyNetworkInterfaceAttribute", fmt.Sprint(err != nil), awsReqStatus(err)).Observe(msSince(start))
 	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
```

Comment:
We have these if statements in multiple places, I am thinking we can instead move all of this inside `BroadcastMissingPermissionsEvent`.

Inside `BroadcastMissingPermissionsEvent` we can check `ok := err.(awserr.Error)` and return if !ok.